tests/ComposerJsonTest.php

```
@@ -0,0 +1,310 @@
+    private function extractPregMatchPattern(string $script): string
+    {
+        $needle = "preg_match('";
+        $start = strpos($script, $needle);
+        $this->assertNotFalse($start, "Expected to find preg_match(' inside script: {$script}");
+
+        $start += strlen($needle);
+        $end = strpos($script, "'", $start);
+        $this->assertNotFalse($end, "Expected a closing quote after the regex pattern in script: {$script}");
+
+        return substr($script, $start, $end - $start);
+    }
```

PHPMD (2.15.0)

[warning] Avoid unused private methods such as 'extractPregMatchPattern'. (undefined)

(UnusedPrivateMethod)